### PR TITLE
CustomBarsOverlayFeature: Add Mana Bank and Health Bank overlay bars

### PR DESCRIPTION
**Pull request by kristofbolyai**
image
image

I already know @magicus won't like https://github.com/Wynntils/Artemis/commit/0445b0626910db1a4ad8f3f722f75012524f1795 . However, there is no other way of doing this, sanely. I think this is our best bet. I made it so if the events are not used, we do not "override" the default minecraft rendering.

### CODE REVIEW

**Comment by magicus** [comment]
I have not played 2.0 nearly enough. Can there only be one "bank" at the time?

**Comment by kristofbolyai** [comment]
> I have not played 2.0 nearly enough. Can there only be one "bank" at the time?

Yes, it is a mage ability upgrade. It is a boss bar, if any mana is present in the bank.

**Comment by magicus** [comment]
You are absolutely correct that I don't like the changes in `BossHealthOverlayMixin`. :-) This will cause incompatibilities with any other mod that tries to modify how boss bars is rendered. I understand if it is tricky to get this right otherwise. If you can hold on a while to this and let me think about it for a while. (Or if you push it, and are prepared to work with me changing the implementation if we figure out a better way.)

**Comment by kristofbolyai** [comment]
@magicus revert last commit and push?

**Comment by kristofbolyai** [comment]
@magicus This should revert the commit you don't like, is it fine to merge otherwise?

**Comment by magicus** [comment]
So this means you can get a new overlay, but you can't cancel the boss bar, so you'll get the same info twice?

If so, well, it's better code-wise, but it's worse UX-wise...

**Comment by kristofbolyai** [comment]
> So this means you can get a new overlay, but you can't cancel the boss bar, so you'll get the same info twice?
> 
> 
> 
> If so, well, it's better code-wise, but it's worse UX-wise...

Yep..

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java [hunk 1]
```diff
@@ -120,25 +142,75 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
...
+                    new GuiScaledOverlaySize(81, 21));
         }
 
         @Override
-        protected void onConfigUpdate(ConfigHolder configHolder) {}
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            if (!WynnUtils.onWorld()) return;
+
+            LerpingBossEvent poolEvent = null;
+            String bloodPoolPercent = "";
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:180]: This is not a rendering thing. Can you extract this into some Util class (one existing, or invent a new if none seem good enough), so other code in the future that wants to read the blood pool value can do so easily? (Same for the Mana Bank)

**Comment by kristofbolyai** [comment]
> So this means you can get a new overlay, but you can't cancel the boss bar, so you'll get the same info twice?
> 
> 
> 
> If so, well, it's better code-wise, but it's worse UX-wise...

Mc only can render 3 boss bars, so I think this limit can be reached (guild info, mob hp, blood pool, shaman mask). This make is so it is seen even then.

**Comment by magicus** [comment]
But yes, if you want to get this in quickly, I can accept it with a double bar, and then we can attack the problem on how to hide a boss bar selectively, in the least intrusive way, as a separate PR.

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:180]: @kristofbolyai Did you miss this, or intentionally skip doing it?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:180]: ooh, I forgot. I will do this later separately.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:180]: I will keep tabs on this and do it.

Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:180]: Thanks! <3

**Pull request merged**